kore/src/Kore/Exec.hs

```diff
@@ -631,13 +648,64 @@ checkFunctions ::
 checkFunctions verifiedModule =
     checkResults $ filter (not . isFunctionPattern . right) equations
   where
-    equations = join $ Map.elems $ extractEquations verifiedModule
+    equations =
+        filter (not . Kore.Equation.isSimplificationRule) $
+            join $ Map.elems $ extractEquations verifiedModule
     -- if any equations fail the check, log the equations and
     -- the entire function returns ExitFailure 3.
     checkResults [] = return ExitSuccess
     checkResults eqns =
         mapM_ errorEquationRightFunction eqns $> ExitFailure 3
 
+{- | Returns true when both equations match the same term.  See:
+ - https://github.com/kframework/kore/issues/2472#issue-833143685
+-}
+bothMatch ::
+    MonadSimplify m =>
+    Equation VariableName ->
+    Equation VariableName ->
+    m Bool
+bothMatch eq1 eq2 =
+    let pre1 = Equation.requires eq1
+        pre2 = Equation.requires eq2
+        arg1 = fromMaybe Predicate.makeTruePredicate $ Equation.argument eq1
+        arg2 = fromMaybe Predicate.makeTruePredicate $ Equation.argument eq2
+        prio1 = fromMaybe Predicate.makeTruePredicate $ Equation.antiLeft eq1
+        prio2 = fromMaybe Predicate.makeTruePredicate $ Equation.antiLeft eq2
+        check =
+            Predicate.makeAndPredicate pre1 $
+                Predicate.makeAndPredicate pre2 $
+                    Predicate.makeAndPredicate arg1 $
+                        Predicate.makeAndPredicate arg2 $
+                            Predicate.makeAndPredicate prio1 prio2
+        check' = Predicate.mapVariables (pure mkConfigVariable) check
+        sort = termLikeSort $ right eq1
+        patt = Pattern.fromPredicateSorted sort check'
+     in (not . isBottom) <$> Pattern.simplify patt
+
+{- | Checks if any function definition in the module carries two equations that both match
+ - same term.
```

Comment:
```suggestion
same term.
```
Haddock will display the `-`. I mistakenly thought it would ignore the dashes when I wrote lines 631-641 for `checkFunctions`. See also line 661.